pkg/pillar/agentlog/cmd/psi-collector/main.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/lf-edge/eve/pkg/pillar/agentlog"
+	"github.com/lf-edge/eve/pkg/pillar/base"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/sirupsen/logrus"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+const (
+	//PIDFile is the file to store the PID
+	PIDFile = types.MemoryMonitorDir + "/psi-collector/psi-collector.pid"
+)
+
+var log *base.LogObject
+
+func createPIDFile() error {
+	f, err := os.Create(PIDFile)
+	if err != nil {
+		log.Errorf("Failed to create PID file: %v", err)
+		return err
+	}
+	defer f.Close()
+	_, err = f.WriteString(fmt.Sprintf("%d", os.Getpid()))
+	if err != nil {
+		log.Errorf("Failed to write PID to file: %v", err)
+		return err
+	}
+	return nil
+}
+
+func main() {
+	logger := logrus.New()
+	log = base.NewSourceLogObject(logger, "psi-collector", os.Getpid())
+	// Check if the collector is already running
+	if _, err := os.Stat(PIDFile); err == nil {
+		log.Noticef("Memory PSI Collector is already running")
+		return
+	}
+	// Create a PID file
+	err := createPIDFile()
```

Comment:
It can hardly be a problem, as the standalone version will be started manually.
But now I remembered that I wanted to detach it from the terminal at the start.